vault_program/src/initialize_vault.rs

```diff
@@ -126,6 +163,87 @@ pub fn process_initialize_vault(
             vault_bump,
             slot,
         )?;
+
+        {
+            // "Mint" the initial VRT supply
+            vault_account.initialize_vault_override_deposit_fee_bps(0, base)?;
+
+            let mint_summary =
+                vault_account.mint_with_fee(initialize_token_amount, initialize_token_amount)?;
+            if mint_summary.vrt_to_depositor != initialize_token_amount
+                || mint_summary.vrt_to_fee_wallet != 0
+            {
+                msg!("Minted VRT to depositor does not match expected amount");
+                return Err(VaultError::VaultInitialAmountFailed.into());
+            }
+
+            vault_account.initialize_vault_override_deposit_fee_bps(deposit_fee_bps, base)?;
+        }
+
+        // Deposit min ST
+        {
+            invoke(
+                &transfer(
+                    token_program.key,
+                    admin_st_token_account.key,
+                    vault_st_token_account.key,
+                    admin.key,
+                    &[],
+                    initialize_token_amount,
+                )?,
+                &[
+                    admin_st_token_account.clone(),
+                    vault_st_token_account.clone(),
+                    admin.clone(),
+                ],
+            )?;
+        }
+
+        // Create ATA
```

Comment:
what're tradeoffs of doing it here vs. caller doing it?

Reply:
also why burn vault over system program?

Reply:
1. We can't create an ATA where the mint does not exist, so it has to be here
2. Absolute worse case scenario, we could recover these funds in case someone fat fingers a large amount of tokens and wants to re-initialize